modules/core/src/main/java/org/apache/synapse/config/SynapsePropertiesLoader.java

```diff
@@ -116,4 +116,25 @@ public static Boolean getBooleanProperty(String name, Boolean def) {
         }
         return Boolean.valueOf(val);
     }
+
+    /**
+     * Get the Integer value of the property from the synapse properties.
+     *
+     * @param name name of the config property
+     * @param def  default value to return if the property is not set
+     * @return the value of the property to be used
+     */
+    public static Integer getIntegerProperty(String name, int def) {
+        String val = getPropertyValue(name, String.valueOf(def));
+        if (val == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Parameter : " + name + " is not defined in the synapse.properties file.");
+            }
+            return def;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("synapse.properties parameter : " + name + " = " + val);
+        }
+        return Integer.valueOf(val);
```

Comment:
Good catch! I'll use `null` instead of `String.valueOf(def)` so the value will return `null` if the property does not exists. With this implementation we can continue printing the debug log line if the property is not set in the `synapse.properties` file

Reply:
> we can continue printing the debug log line if the property is not set in the synapse.properties file

That have been already handled as well in the underlying method (`MiscellaneousUtil#getProperty` [1] ), it seems!

[1] https://github.com/heysulo/wso2-synapse/blob/63e4bc4bcf1005f5dcdd026a3ac020a3f70626b9/modules/commons/src/main/java/org/apache/synapse/commons/util/MiscellaneousUtil.java#L64

Reply:
Right! What you suggested is the ideal implementation. I'll correct it

Reply:
Corrected!